client/src/daemon_launcher.cpp

```diff
@@ -38,7 +44,51 @@ pid_t daemon_launcher::start() {
     return ret;
 }
 
-#ifndef __APPLE__
+#include <iostream>
+
+#ifdef __APPLE__
+void daemon_launcher::open(simpleipc::client::service_client_impl& impl) {
+    remove(service_path.c_str());
+
+    int kq = kqueue();
+    int f = ::open(FileUtil::getParent(service_path).c_str(), O_RDONLY);
+
+    pid_t proc = start();
+
+    struct kevent k_ev;
+    EV_SET(&k_ev, f, EVFILT_VNODE, EV_ADD | EV_CLEAR, NOTE_WRITE, 0, NULL);
+    kevent(kq, &k_ev, 1, NULL, 0, NULL);
+
+    EV_SET(&k_ev, proc, EVFILT_PROC, EV_ADD | EV_CLEAR, NOTE_EXIT, 0, NULL);
+    kevent(kq, &k_ev, 1, NULL, 0, NULL);
```

Comment:
This could be merged into one kqueue call?

Reply:
Nah, you can't do that.